### ISSUE TITLE
import: invalid code generated

### ISSUE THREAD

**Issue by TealG** [capnproto/capnp-ocaml#46]
When an imported type is only referenced in a return value, invalid code is generated.
Example:

```
using Test = import "test.capnp";

interface TestImportInterface {
  foo @0 () -> (result :Test.TestAllTypes);
}
```

The generated code does not actually import `Test`, but references it like:

```
...
          val result_get : t -> TestAllTypes.t
          val result_set_reader : t -> TestAllTypes.struct_t reader_t -> TestAllTypes.t
          val result_set_builder : t -> TestAllTypes.t -> TestAllTypes.t
          val result_init : t -> TestAllTypes.t
...
```

Which does not compile:

```
Error: Unbound module TestAllTypes
```

However, if there's another reference to the imported type e.g. in a struct field, it results in different code that compiles:

```
using Test = import "test.capnp";

struct TestImport {
  field @0 :Test.TestAllTypes;
}

interface TestImportInterface {
  foo @0 () -> (result :Test.TestAllTypes);
}
```

Generated code:

```
...
          val result_get : t -> [`TestAllTypes_a0a8f314b80b63fd] builder_t
          val result_set_reader : t -> [`TestAllTypes_a0a8f314b80b63fd] reader_t -> [`TestAllTypes_a0a8f314b80b63fd] builder_t
          val result_set_builder : t -> [`TestAllTypes_a0a8f314b80b63fd] builder_t -> [`TestAllTypes_a0a8f314b80b63fd] builder_t
          val result_init : t -> [`TestAllTypes_a0a8f314b80b63fd] builder_t
...
````

Tested with v3.1.0 (ocaml 4.05.0)

**Comment by TealG** [comment]
@talex5 any idea where the issue might be?

**Comment by talex5** [comment]
The output is generated by https://github.com/capnproto/capnp-ocaml/blob/master/src/compiler/generate.ml, so probably best to start looking there.

Maybe `GenCommon.filter_interesting_imports` has something to do with this?
https://github.com/capnproto/capnp-ocaml/blob/master/src/compiler/generate.ml#L233

**Comment by talex5** [comment]
Fixed in #65.
